allennlp/tests/version_test.py

```diff
@@ -0,0 +1,52 @@
+import re
+
+import pytest
+
+from allennlp.version import VERSION
+
+
+# Regex to check that the current version set in `allennlp.version` adheres to
+# PEP 440, as well as some of our own internal conventions, such as the `.dev`
+# suffix being used only for nightly builds.
+# 0.0.0rc0.post0.dev20200424
+VALID_VERSION_RE = re.compile(
+    r"^"
+    r"(0|[1-9]\d*)"  # major
```

Comment:
Is this the right regex?  Isn't the `* just on the `\d` rather than on the whole expression?

Reply:
It is just on the `\d`, which I think gives the correct behavior because it disallows a preceding "0", unless by itself. For example:

- ✅  "0"
- ✅  "1"
- ✅  "10"
- ❌  "01"

Reply:
sorry--brain fart